src/main/java/org/jenkinsci/plugins/runselector/selectors/DisplayNameRunSelector.java

```diff
@@ -0,0 +1,50 @@
+package org.jenkinsci.plugins.runselector.selectors;
+
+import hudson.Extension;
+import hudson.model.Job;
+import hudson.model.Run;
+import org.jenkinsci.Symbol;
+import org.jenkinsci.plugins.runselector.RunSelectorDescriptor;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Selects the build based on its display name.
+ *
+ * @author Alexandru Somai
+ */
+public class DisplayNameRunSelector extends AbstractSpecificRunSelector {
+
+    @DataBoundConstructor
+    public DisplayNameRunSelector(String runDisplayName) {
+        super(runDisplayName);
+    }
+
+    @Nonnull
+    public String getRunDisplayName() {
+        return getParameter();
+    }
+
+    @Override
+    public Run<?, ?> getBuild(@Nonnull Job<?, ?> job, @Nonnull String resolvedParameter) {
+        for (Run<?, ?> build : job.getBuilds()) {
```

Comment:
It is also ineffective from the performance PoV since we lazy load builds. Are they sorted in the descending order here? I hope so